torch/csrc/api/include/torch/nn/functional/pooling.h

```diff
@@ -776,8 +776,8 @@ inline std::tuple<Tensor, Tensor> fractional_max_pool2d_with_indices(
 
   Tensor _random_samples_ = _random_samples;
   if (!_random_samples_.defined()) {
-    auto n_batch = input.dim() == 3;
-    _random_samples_ = torch::rand({n_batch, input.size(-1), 2}, torch::TensorOptions().dtype(input.dtype()).device(input.device()));
+    auto n_batch = input.dim() == 3 ? 1 : input.size(0);
+    _random_samples_ = torch::rand({n_batch, input.size(-3), 2}, torch::TensorOptions().dtype(input.dtype()).device(input.device()));
```

Comment:
Double checked with @thomasjpfan that -3 should be the correct index (since it maps to `input.size(0)`, which now agrees with the Python implementation